compiler/control/OMROptions.cpp

```diff
@@ -2217,6 +2218,14 @@ OMR::Options::jitLatePostProcess(TR::OptionSet *optionSet, void * jitConfig)
             if (_coldUpgradeSampleThreshold == TR_DEFAULT_COLD_UPGRADE_SAMPLE_THRESHOLD)
                _coldUpgradeSampleThreshold = 2;
             }
+         else
+            {
+            // For AOT with GCR enabled we can be more conservative with 
+            // upgrades through sampling because we can rely on GCR
+            if (!self()->getOption(TR_DisableGuardedCountingRecompilations) &&
+              _coldUpgradeSampleThreshold == TR_DEFAULT_COLD_UPGRADE_SAMPLE_THRESHOLD)
+               _coldUpgradeSampleThreshold = 10;
```

Comment:
Is the purpose here to reduce the likelihood of sampling causing upgrade compilations during startup ? i.e. GCR would be expected to cause many more upgrades outside of startup anyway.

Reply:
Could we maybe add a comment to explain `10` ?